### DISCUSSION TITLE
Increasing output power for USRP B210

### DISCUSSION THREAD

**Discussion by aibtw** [srsran/srsRAN_Project#321] [General Help] [answered]
Hi, 

I am using the gnb from srsRAN project with USRP B210. I tried to increase the gain from the configuration file like so:
```
amf:
  addr: 127.0.1.100                                             # The address or hostname of the AMF.
  bind_addr: 127.0.0.1                                          # A local IP that the gNB binds to for traffic from the AMF.

ru_sdr:
  device_driver: uhd                                            # The RF driver name.
  device_args: type=b200,num_recv_frames=64,num_send_frames=64  # Optionally pass arguments to the selected RF driver.
  clock: internal                                               # Specify the clock source used by the RF.
  sync:                                                         # Specify the sync source used by the RF.
  srate: 23.04                                                  # RF sample rate might need to be adjusted according to selected bandwidth.
  otw_format: sc12
  tx_gain: 89                                                   # Transmit gain of the RF might need to adjusted to the given situation.
  rx_gain: 76                                                   # Receive gain of the RF might need to adjusted to the given situation.

cell_cfg:
  dl_arfcn: 627340                                              # ARFCN of the downlink carrier (center frequency).
  band: 78                                                      # The NR band.
  channel_bandwidth_MHz: 20                                     # Bandwith in MHz. Number of PRBs will be automatically derived.
  common_scs: 30                                                # Subcarrier spacing in kHz used for data.
  plmn: "99970"                                                 # PLMN broadcasted by the gNB.
  tac: 7                                                        # Tracking area code (needs to match the core configuration).
  pci: 1                                                        # Physical cell ID.
  nof_antennas_dl: 2
  nof_antennas_ul: 2
```
is the tx_gain and rx_gain in db? because we know 89 db tx gain is the maximum supported by usrp B210:

> All frontends have individual analog gain controls. The receive frontends have 76 dB of available gain; and the transmit frontends have 89.8 dB of available gain. Gain settings are application specific, but it is recommended that users consider using at least half of the available gain to get reasonable dynamic range. 

However, I tried to measure the power at the tx port of the USRP, and it shows at most -60 dBm.
This is extremely low for my case. I am trying to get something like -20 to -10 dBm. 

Am I missing something? would appreciate the help.

**Comment by fllay** [comment]
How do you measure the power? Using power meter? Most of the power meter measure heat from the sensor so you get the average power over TDD slots that why it is low. I would suggest you to use a spectrum analyzer that can gate the input according to TDD frame and you can determine the power of the PSD.

**Reply by aibtw** [reply]
No, I actually got the -60dBm using a spectrum analyzer.

**Reply by fllay** [reply]
Can you show the power spectrum density from your spsectrum analyzer?

**Comment by robertfalkenberg** [comment, marked as answer]
Also note to measure during a full-load downlink transmission. Otherwise the gNB only transmits very little power.

**Reply by aibtw** [reply]
I am pretty sure I tested it with high load downlink transmission, but let me try again just to make sure. 

Also, is the "tx_gain" configuration the only thing I have to change or is there another config that I may have missed?

**Reply by robertfalkenberg** [reply]
For USRPs this is the only setting.

**Comment by yc541** [comment]
We measured the output of a B205mini (max supported tx_gain) with a power meter while running srsRAN_4G on an FDD band, and it was about -15 to -12 dBm, maybe try an FDD band as well?
